src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -369,6 +380,45 @@ func (r *Cluster) validatePandaproxyListeners() field.ErrorList {
 	return allErrs
 }
 
+func (r *Cluster) validateSchemaRegistryListener() field.ErrorList {
+	var allErrs field.ErrorList
+	schemaRegistry := r.Spec.Configuration.SchemaRegistry
+
+	if r.IsSchemaRegistryExternallyAvailable() {
+		kafkaExternal := r.ExternalListener()
+
+		if kafkaExternal == nil || !kafkaExternal.External.Enabled {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("schemaRegistry"),
+					r.Spec.Configuration.SchemaRegistry,
+					"cannot have a schema registry external listener without a kafka external listener"))
+		}
+		if kafkaExternal == nil && schemaRegistry.External.Subdomain != "" {
+			allErrs = append(allErrs,
+				field.Invalid(field.NewPath("spec").Child("configuration").Child("schemaRegistry").Child("external").Child("subdomain"),
+					r.Spec.Configuration.SchemaRegistry.External.Subdomain,
+					"kafka external listener is empty but must specify the same subdomain as that of the external schema registry"))
```

Comment:
I think a more straightforward message would be
```suggestion
					"the external kafka listener can't be empty if the registry subdomain is set"))
```

Reply:
✅